internal/discovery/foreign-cluster-operator/foreign-cluster-controller.go

```diff
@@ -473,7 +437,70 @@ func (r *ForeignClusterReconciler) checkJoined(fc *discoveryv1alpha1.ForeignClus
 // check if the error is due to a TLS certificate signed by unknown authority
 func isUnknownAuthority(err error) bool {
 	var err509 x509.UnknownAuthorityError
-	return goerrors.As(err, &err509)
+	var err509Hostname x509.HostnameError
+	return goerrors.As(err, &err509) || goerrors.As(err, &err509Hostname)
+}
+
+func (r *ForeignClusterReconciler) getHomeAuthUrl() (string, error) {
+	address, _ := os.LookupEnv("APISERVER")
```

Comment:
Can we use the default kubernetes service environment variable?

Reply:
no, this variable has an internal IP of the service, but here we need an address that can be reachable from the foreign cluster